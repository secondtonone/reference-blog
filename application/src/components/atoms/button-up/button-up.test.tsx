import renderer from 'react-test-renderer';
import ButtonUp from '.';
import SvgIcon from '~/components/containers/svg-icon';

jest.mock('~/components/containers/svg-icon', () => jest.fn().mockReturnValue('svg-icon'));

describe('Match snapshots ButtonUp', () => {
  it('default', () => {
    const tree = renderer.create(<ButtonUp />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(SvgIcon).toHaveBeenCalledWith({ code: 'up', size: [16, 16] }, {});
  });
  it('active', () => {
    const tree = renderer.create(<ButtonUp active />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(SvgIcon).toHaveBeenCalledWith({ code: 'up', size: [16, 16] }, {});
  });
});
