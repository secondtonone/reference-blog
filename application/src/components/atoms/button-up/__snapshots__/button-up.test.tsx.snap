// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots ButtonUp active 1`] = `
<span
  className="button-up__UpStyled-sc-yq92rm-0 irGsTM"
>
  <span
    className="base-button__BaseButton-sc-1hqbwl-0 primary-button__PrimaryButton-sc-1gi77ac-0 dGmWiH bZuBBR button-icon__ButtonIconStyled-sc-i4ys4z-0 hCbfec"
    data-test="button-icon-up"
    size="m"
  >
    <button
      aria-label="To Top"
      className="___SButton_rkzvx_gg_ _size_m_rkzvx_gg_ _theme_primary-info_rkzvx_gg_ __active_rkzvx_gg_"
      data-ui-name="Button"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      size="m"
      style={Object {}}
      tabIndex={0}
      type="button"
    >
      <span
        className="___SInner_rkzvx_gg_"
        data-ui-name="Box"
        style={Object {}}
      >
        <span
          className="_size_m_rkzvx_gg_ ___SText_rkzvx_gg_"
          data-ui-name="Button.Text"
          size="m"
          style={Object {}}
        >
          svg-icon
        </span>
      </span>
    </button>
  </span>
</span>
`;

exports[`Match snapshots ButtonUp default 1`] = `
<span
  className="button-up__UpStyled-sc-yq92rm-0 irGsTM"
>
  <span
    className="base-button__BaseButton-sc-1hqbwl-0 secondary-button__SecondaryButton-sc-1lq79t2-0 dGmWiH PcQft button-icon__ButtonIconStyled-sc-i4ys4z-0 czAHmw"
    data-test="button-icon-up"
    size="m"
  >
    <button
      aria-label="To Top"
      className="___SButton_rkzvx_gg_ _size_m_rkzvx_gg_ _theme_secondary-muted_rkzvx_gg_"
      data-ui-name="Button"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      size="m"
      style={Object {}}
      tabIndex={0}
      type="button"
    >
      <span
        className="___SInner_rkzvx_gg_"
        data-ui-name="Box"
        style={Object {}}
      >
        <span
          className="_size_m_rkzvx_gg_ ___SText_rkzvx_gg_"
          data-ui-name="Button.Text"
          size="m"
          style={Object {}}
        >
          svg-icon
        </span>
      </span>
    </button>
  </span>
</span>
`;
